SocialCareCaseViewerApi/V1/UseCase/PersonalRelationshipsUseCase.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Exceptions;
+using SocialCareCaseViewerApi.V1.Gateways;
+using SocialCareCaseViewerApi.V1.UseCase.Interfaces;
+
+namespace SocialCareCaseViewerApi.V1.UseCase
+{
+    public class PersonalRelationshipsUseCase : IPersonalRelationshipsUseCase
+    {
+        private IDatabaseGateway _databaseGateway;
+
+        public PersonalRelationshipsUseCase(IDatabaseGateway databaseGateway)
+        {
+            _databaseGateway = databaseGateway;
+        }
+
+        public void ExecutePost(CreatePersonalRelationshipRequest request)
+        {
+            var persons = _databaseGateway.GetPersonsByListOfIds(new List<long>() { request.PersonId, request.OtherPersonId });
+
+            var personDoesNotExist = persons.Find(person => person.Id == request.PersonId) == null;
+            if (personDoesNotExist) throw new PersonNotFoundException($"\"personId\" with \"{request.PersonId}\" was not found.");
+
+            var otherPersonDoesNotExist = persons.Find(person => person.Id == request.OtherPersonId) == null;
+            if (otherPersonDoesNotExist) throw new PersonNotFoundException($"\"otherPersonId\" with \"{request.OtherPersonId}\" was not found.");
+
+            var type = _databaseGateway.GetPersonalRelationshipTypeByDescription(request.Type);
+
+            var typeDoesNotExist = type == null;
+            if (typeDoesNotExist) throw new PersonalRelationshipTypeNotFoundException($"\"type\" with \"{request.Type}\" was not found.");
+
+            var personWithPersonalRelationships = _databaseGateway.GetPersonWithPersonalRelationshipsByPersonId(request.PersonId);
+
+            var personalRelationships = personWithPersonalRelationships.PersonalRelationships;
+            var personalRelationshipAlreadyExists = personalRelationships.Find(pr => pr.OtherPersonId == request.OtherPersonId && pr.Type.Description == request.Type) != null;
+            if (personalRelationshipAlreadyExists) throw new PersonalRelationshipAlreadyExistsException($"Personal relationship with \"type\" of \"{request.Type}\" already exists.");
+
+            _databaseGateway.CreatePersonalRelationship(request);
+
+            _databaseGateway.CreatePersonalRelationship(new CreatePersonalRelationshipRequest()
+            {
+                PersonId = request.OtherPersonId,
+                OtherPersonId = request.PersonId,
+                TypeId = type.InverseTypeId,
+                IsMainCarer = null,
+                IsInformalCarer = null,
+                Details = null
```

Comment:
The CreatePersonalRelationshipRequest object states these values can't be null.

You could make them nullable but keep the FluentValidation checks to ensure the actual API request validates they aren't null. This way you know the request actually has assigned the values but can also set the value as null directly yourself without compiler warnings.

Reply:
That makes sense, I think I keep going confused by `null!`. Could you check that the latest commit is correct please?

Reply:
So if something is nullable (got the ? after the type) we don't need `null!`.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving

My understanding is saying `null!` is basically telling the compiler I know this isn't meant to be null but I know right now I'm not assigning this a value so for now it is null and that's ok.

Reply:
Ahhh okay, I got! Your explanation is much clearer than the docs for me. 🙇‍♀️

Will remove `null!`.